rotkehlchen/api/v1/schemas.py

```diff
@@ -517,6 +517,21 @@ class TradeSchema(Schema):
     link = fields.String(load_default=None)
     notes = fields.String(load_default=None)
 
+    @validates_schema
+    def validate_fee_and_fee_currency(  # pylint: disable=no-self-use
+        self,
+        data: Dict[str, Any],
+        **_kwargs: Any,
+    ) -> None:
+        fee = data.get('fee')
+        fee_currency = data.get('fee_currency')
+
+        if not ((fee and fee_currency) or not (fee or fee_currency)):
```

Comment:
This condition is really complicated and hard to read. I had to write a test to make sure it's correct :sweat: 

Can you please simplify it?

```suggestion
        if not fee or not fee_currency and (fee or fee_currency):
```

Also perhaps explain in the docstring of the `validate_fee_and_fee_currency` function what it checks for.

Reply:
So I replaced the logic with something more readable
```py
if not all([fee, fee_currency]) and any([fee, fee_currency]):
    ...
```

Although the logic operator was faster, this is supposedly more readable and easier to understand.
- Using logic operators only  --->`0.035516125`
- Using `any` and `all` ---> `0.201628292`

Reply:
Normally I would go for speed especially with such a big difference of almost 6x. But we can adjust if this becomes a bottleneck.

Adding the comment already helps a lot with readability.